integration/hive/src/main/java/org/apache/carbondata/hive/Hive2CarbonExpression.java

```diff
@@ -103,38 +111,43 @@ public static Expression convertExprHive2Carbon(ExprNodeDesc exprNodeDesc) {
               getDateType(l1.get(left).getTypeString()));
         }
         LiteralExpression literalExpression =
-            new LiteralExpression(l1.get(right).getExprString().replace("'", ""),
+            new LiteralExpression(getExpressionValue(l1.get(right)),
                 getDateType(l1.get(right).getTypeString()));
         return new EqualToExpression(columnExpression, literalExpression);
       } else if (udf instanceof GenericUDFOPEqualOrGreaterThan) {
         ColumnExpression columnExpression = new ColumnExpression(l1.get(left).getCols().get(left),
             getDateType(l1.get(left).getTypeString()));
-        LiteralExpression literalExpression = new LiteralExpression(l1.get(right).getExprString(),
-            getDateType(l1.get(left).getTypeString()));
+        LiteralExpression literalExpression =
```

Comment:
can we make two methods ?
`makeColumnExpression(l1)
``makeLiteralExpression(l1)`

I see lot of duplicate code here

Reply:
i dont think its good, we would just be replacing 1 line code with a method call.